src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1202,16 +1195,6 @@ internal virtual bool TryApplyChanges(Solution newSolution, IProgressTracker pro
                     return false;
                 }
 
-                // make sure that newSolution is a branch of the current solution
-
-                // the given solution must be a branched one.
-                // otherwise, there should be no change to apply.
-                if (oldSolution.BranchId == newSolution.BranchId)
-                {
-                    CheckNoChanges(oldSolution, newSolution);
-                    return true;
-                }
```

Comment:
also not certain what teh value here is.  If someone asks us to apply changes, we can simply walk through and figure out waht the changes were.  This was a fast-bail-out for a case that seems like the unlikely one (e.g. when does someone ask to apply the same solution over the exisitng solution)?